internal/tokens/oauth.go

```diff
@@ -66,19 +67,44 @@ func combinedClientInfoHandler(r *http.Request) (clientID, clientSecret string,
 	return
 }
 
+func isUriValid(clientHost, redirectHost string) bool {
+	clientHostParts := strings.Split(clientHost, ".")
+	redirectHostParts := strings.Split(redirectHost, ".")
+
+	if len(clientHostParts) != len(redirectHostParts) {
+		return false
+	}
+
+	for i := range clientHostParts {
+		if clientHostParts[i] == "*" {
+			continue
+		}
+		if clientHostParts[i] != redirectHostParts[i] {
```

Comment:
Minor: I would swap the args - because we are comparing against clientHostParts (currently it is a yoda condition)
```suggestion
		if redirectHostParts[i] != clientHostParts[i] {
```